_source/_posts/2022-09-09-supabase-vue.md

```diff
@@ -0,0 +1,941 @@
+            :date="entryDate"
+            :onSaved="saveEntry"
+            :onCancelled="endAddEntry"
+          />
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      month: parseInt(this.initialMonth),
+      year: parseInt(this.initialYear),
+      numberOfDaysInMonth: 0,
+      showModal: false,
+      entryDate: undefined,
+      entries: [],
+      showEntryDetailModal: false,
+      selectedEntry: undefined,
+    }
+  },
+  computed: {
+    numberOfDaysInMonth() {
+      return new Date(this.year, this.month + 1, 0).getDate()
+    },
+    firstWeekSkipDays() {
+      return new Date(this.year, this.month, 1).getDay()
+    },
+  },
+  async mounted() {
+    await this.getEntries()
+  },
+  methods: {
+    async getEntries() {
+      const startdate = new Date(this.year, this.month, 1)
+      const endDate = new Date(this.year, this.month + 1, 1)
+
+      const query = supabase
+        .from('Entries')
+        .select()
+        .gte('date', startdate.toLocaleString())
+        .lt('date', endDate.toLocaleString())
+
+      if (!this.authState?.isAuthenticated) {
+        query.eq('isPublic', true)
+      }
+      const { data, error } = await query
+      this.entries = data
+    },
+    async previousMonth() {
+      this.year = this.month === 0 ? this.year - 1 : this.year
+      this.month = (((this.month - 1) % 12) + 12) % 12
+
+      await this.getEntries()
+    },
+    async nextMonth() {
+      this.year = this.month === 11 ? this.year + 1 : this.year
+      this.month = (this.month + 1) % 12
+
+      await this.getEntries()
+    },
+    addEntry(date) {
+      this.entryDate = date
+      this.showModal = true
+    },
+    async saveEntry(entry) {
+      const { data, error } = await supabase.from('Entries').upsert(entry)
+
+      if (data) {
+        this.entries.push(data[0])
+      }
+
+      this.endAddEntry()
+    },
+    endAddEntry() {
+      this.entryDate = undefined
+      this.showModal = false
+    },
+    showEntryDetail(entryId) {
+      this.selectedEntry = this.entries.filter((r) => r.id === entryId)[0]
+      this.showEntryDetailModal = true
+    },
+    hideEntryDetail() {
+      this.showEntryDetailModal = false
+      this.selectedEntry = undefined
+    },
+  },
+  props: {
+    initialMonth: Number,
+    initialYear: Number,
+  },
+}
+</script>
+```
+{% endraw %}
+
+This component displays a calendar vue for any entries. It defaults to the current month but the user can move back and forth in time to change the month. The calendar also has the responsibility of populating its data. This means you will import the supabase client you created earlier and use it to fetch the data from your supabase project.
+
+The component will check if the user is authenticated before querying the data. If the user is authenticated the component will request all entries, however, if they are not then it will only request public entries.
```

Comment:
```suggestion
The component will check if the user is authenticated before querying the data. If the user is authenticated the component will request all entries.  However, if they are not authenticated, it will only request public entries.
```